include/miopen/miopen.h

```diff
@@ -1951,16 +1984,27 @@ miopenConvolutionBackwardData(miopenHandle_t handle,
 
 /*! @brief Get the GPU memory required for the backward weights convolution algorithm.
  *
+ * For given tensor and convolution descriptors, this function calculates and returns the minimum
+ * size of the workspace that must be provided to miopenFindConvolutionBackwardWeightsAlgorithm() in
+ * order for the latter to find the best candidate from the available backward weights convolution
+ * algorithms.
  *
- * For a provided tensor descriptors and algorithm selection, this function calculates and returns
- * the workspace size required for back propagation on data. This call is required and must be
- * executed once before running miopenFindConvolutionBackwardWeightsAlgorithm() in order to
- * determine
- * the largest required allocation for the algorithm search; i.e., the maximum size of the memory
- * needed from the set of potential backward weights convolution algorithm is returned.
+ * It should be assumed that the required workspace size is different for each convolution
+ * configuration. Therefore, typically this function should be called at least once for each
+ * convolution configuration used.
  *
- * If using Group/Depthwise convolution mode, call miopenSetConvolutionGroupCount() before running
- * this.
+ * Since the convolution configuration is determined by tensor and convolution descriptors, the user
+ * should ensure that all descriptors contain complete information. For example, if Group/Depthwise
+ * convolution mode is used, then miopenSetConvolutionGroupCount() should be called before running
+ * this, and so on.
+ *
+ * WARNING: Providing smaller workspace may result in the selection of a slow convolution
+ * algorithm, and therefore affect library performance. Moreover, sub-optimal data may be
+ * cached in the user's find-db. This means that the performance drop will become persistent,
+ * i.e. even providing sufficient workspace won't restore the performance. To get rid of
+ * this problem, the user will need to either remove the user's find-db, or repeat
+ * miopenFindConvolutionBackwardWeightsAlgorithm() with affected convolution configs
+ * in Normal Find Mode (the latter will overwrite sub-optimal user's find-db records).
```

Comment:
@tbennun 
> Moreover, sub-optimal data may be cached in the user's find-db. This means that the performance drop will become persistent, i.e. even providing sufficient workspace won't restore the performance.

This is something that I really don't like. I am going to disable caching of possibly sub-optimal entries to find-db (and provide an option to enable it if absolutely needed). What do you think?

/cc @JehandadKhan @junliume @averinevg

Reply:
@tbennun@JehandadKhan @averinevg Implemented in https://github.com/ROCm/MIOpen/pull/2947/files/2a5802d156e76a779e5f85207e528eae7c3a1dc7..a610dd16c89deae366ebae6cad926d2bdb6ad46c

Reply:
It's fine with me since users can enable it with an environment variable. Thanks!